responsibleai/responsibleai/_managers/causal_manager.py

```diff
@@ -2,30 +2,168 @@
+
+        if causal_config.is_duplicate(self._causal_config_list):
+            raise DuplicateManagerConfigException("Duplicate causal configuration detected.")
+
+        self._causal_config_list.append(causal_config)
 
     def compute(self):
-        pass
+        """Computes the causal effects by running the causal configuration."""
+        for config in self._causal_config_list:
+            if not config.is_computed:
+                config.is_computed = True
+                try:
+                    if config.nuisance_model not in [CausalConstants.AUTOML, CausalConstants.LINEAR]:
+                        raise UserConfigValidationException(
+                            "nuisance_model should be one of ['{}', '{}'], got {}".format(
+                                CausalConstants.AUTOML, CausalConstants.LINEAR, config.nuisance_model))
 
-    def get(self):
-        raise NotImplementedError("Get not implemented for CausalManager")
+                    is_classification = self._task_type == ModelTask.CLASSIFICATION
+                    causal_analysis = CausalAnalysis(config.features, config.categorical_features,
+                                                     heterogeneity_inds=config.treatment_features,
+                                                     classification=is_classification,
+                                                     nuisance_models=config.nuisance_model,
+                                                     n_jobs=-1)
+                    X = self._data.drop([self._target_column], axis=1)
+                    y = self._data[self._target_column].values
+                    causal_analysis.fit(X, y)
+                    config.causal_effects = causal_analysis.global_causal_effect(alpha=0.05)
+                except Exception as e:
+                    config.has_computation_failed = True
+                    config.failure_reason = str(e)
+
+
+    def get(self, failed_to_compute=False):
+        """
+        Get the computed causal effects or failure reason.
+
+        :param failed_to_compute: Get the failure reasons if causal effects
+            failed to compute.
+        :type failed_to_compute: bool
+        """
+        if not failed_to_compute:
+            causal_effects = []
+            for config in self._causal_config_list:
+                if config.is_computed and not config.has_computation_failed:
+                    causal_effects.append(config.causal_effects)
+            return causal_effects
+        else:
+            failure_reason_list = []
+            for config in self._causal_config_list:
+                if config.is_computed and config.has_computation_failed:
+                    failure_reason_list.append(config.failure_reason)
+            return failure_reason_list
 
     def list(self):
         pass
 
     @property
     def name(self):
-        """Get the name of the causal manager.
+        """
+        Get the name of the causal manager.
```

Comment:
I don't think this is preferred style?  I think the comment should follow """.
That's how it is elsewhere in the code.

Reply:
Done, thanks